flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/ExecutionGraph.java

```diff
@@ -694,6 +695,18 @@ public void cancel() {
 		}
 	}
 
+	public void stop() {
+		if(jobType == JobType.STREAMING) {
+			for(ExecutionVertex ev : this.getAllExecutionVertices()) {
+				if(ev.getNumberOfInputs() == 0) { // send signal to sources only
+					ev.stop();
+				}
+			}
+			return;
+		}
+		throw new RuntimeException("STOP is only supported by streaming jobs.");
```

Comment:
I think it's better to write `if () {...} else { throw ... }`. This will get rid off the disrupted control flow here.